### PR TITLE
Added test cases for inputs with errors during translation

### PR DESCRIPTION
**Pull request by itssamuelrowe**
The translator spec was updated to ensure that input with syntax and semantic errors are properly reported when `--emit-lua` is specified. It should be noted that the test spec does not actually invoke the compiler with the flag, instead it invokes an internal
function to simulate it.

### CODE REVIEW

File: pallene/parser.lua [hunk 1]
```diff
@@ -249,8 +249,8 @@ local grammar = re.compile([[
     toplevelrecord  <- (P  RECORD NAME^NameRecord recordfields
                            END^EndRecord)                        -> ToplevelRecord
 
-    export_or_local <- LOCAL -> to_true
-                     / EXPORT -> to_false
+    export_or_local <- (LOCAL -> to_true
+                     / EXPORT -> to_false)
```
Comment (hugomg) [inline pallene/parser.lua:253]: Why are we adding `()` here?

File: pallene/driver.lua [hunk 2]
```diff
@@ -125,6 +126,18 @@ end
 --    compile("c", "so", "foo.c)      --> outputs "foo.so"
 --
 
+function driver.translate(input_file_name, input)
```
Comment (hugomg) [inline pallene/driver.lua:129]: I think it is a bit strange that the test suite is using different code paths for the cases that give errors and the cases that don't. In the former we invoke pallenec and in the latter we call a special function from driver.lua.

What do you think about calling pallenec in both cases?

```
local function compile(pallene_code)
    assert(util.set_file_contents("__test__.pln", pallene_code))
    local ok, _, _, error_message = util.outputs_of_execute("./pallenec __test__.pln --emit-lua")
    return ok, error_message
end

local function assert_translation(pallene_code, expected)
    assert(compile(pallene_code))
    local contents = util.get_file_contents("__test__.lua")
    assert.are.same(expected, contents)
end

local function assert_translation_error(pallene_code, expected)
    local ok, err = compile("__test__.pln", pallene_code)
    assert.is.falsy(ok)
    assert.match(expected, err, 1, true)
end
```

File: pallene/driver.lua [hunk 3]
```diff
@@ -50,6 +50,7 @@ function driver.compile_internal(filename, input, stop_after)
     end
 
     prog_ast, errs = checker.check(prog_ast)
+
```
Comment (hugomg) [inline pallene/driver.lua:53]: Do we need this empty line?

File: pallene/syntax_errors.lua [hunk 4]
```diff
@@ -175,6 +175,8 @@ local errors = {
     AssignNotToVar = "Expected a valid lvalue in the left side of assignment but found a regular expression.",
 
     CastMissingType = "Expected a type for the cast expression.",
+
+    LocalOrExportRequired = "Function definition must include 'local' or 'export' modifier."
```
Comment (hugomg) [inline pallene/syntax_errors.lua:179]: Is this related to #261 or is it something else?

Comment (itssamuelrowe) [reply pallene/syntax_errors.lua]: No, it's not related to #261. I thought it would be nice to have a custom message is all. I've removed it.

File: pallene/driver.lua [hunk 5]
```diff
@@ -133,14 +145,12 @@ local function compile_pln_to_lua(input_ext, output_ext, input_file_name, base_n
         return false, { err }
     end
 
-    local prog_ast, errs = driver.compile_internal(input_file_name, input, "checker")
-    if not prog_ast then
+    local translation, errs = driver.translate(input_file_name, input)
+    if not translation then
```
Comment (hugomg) [inline pallene/driver.lua:149]: I think we can keep all the logic inside compile_pln_to_lua like it was before. No need to split part of it into an `driver.translate` exported function.

File: spec/translator_spec.lua [hunk 6]
```diff
@@ -1,4 +1,5 @@
 local util = require "pallene.util"
+local driver = require "pallene.driver"
```
Comment (hugomg) [inline spec/translator_spec.lua:2]: The linter is compaining about this unused variable.

File: spec/translator_spec.lua [hunk 7]
```diff
@@ -14,11 +15,58 @@ local function assert_translation(pallene_code, expected)
...
+    assert.is_false(ok)
+    assert.match(expected, actual, 1, true)
+end
+
 local function cleanup()
     os.remove("__test__.pln")
     os.remove("__test__.lua")
 end
 
+describe("Syntax errors during Pallene to Lua translation", function ()
```
Comment (hugomg) [inline spec/translator_spec.lua:30]: We could replace each of these describe blocks with a single test, as discussed on Slack.

File: pallene/driver.lua [hunk 8]
```diff
@@ -134,13 +134,14 @@ local function compile_pln_to_lua(input_ext, output_ext, input_file_name, base_n
     end
 
     local prog_ast, errs = driver.compile_internal(input_file_name, input, "checker")
+
```
Comment (hugomg) [inline pallene/driver.lua]: Can we remove this line?
